inc/Classes/File.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace LanSuite;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Filesystem\Path;
+/**
+ * Class File
+ * Generic class for file accessor functions
+ */
+
+class File
+{
+    private static string $_basePath='';
+    private string $_relativePath='';
+    private Filesystem $_fileSystem;
+
+    /**
+     * Constructor sets base path and initializes FS object
+     * 
+     * @param Filesystem $fs Filesystem-object for dependency injection
+     */
+    public function __construct(Filesystem $fs = new Filesystem)
+    {
+        $this->_fileSystem = $fs;
+        self::$_basePath = Path::canonicalize(ROOT_DIRECTORY);
```

Comment:
Can we also inject `ROOT_DIRECTORY` as a function default argument? This makes unit testing easier.

Reply:
Yes and maybe.
For unit testing purposes it makes sense, from the intent of the class it may not.
As that is to ensure that no file access happens below the root directory in all use cases.
If the root directory can be modified with the intialisation, then that has the risk of improper use.

Reply:
Why not provding a `public static` function that does set the respective values?
This way, this can be used in both ways.

Reply:
that has the risk of being used to set a different basepath for one object instance, but affecting all of them instead.
Unless we ensure that basepath is only set once during intialization and cannot be modified after that (actually did that in the first iteration and had a static bool `basePathInitialised` for that).
Fine for me to go that way (...again)